packages/tx/src/eip1559Transaction.ts

```diff
@@ -0,0 +1,353 @@
+import { BN, bnToHex, bnToRlp, ecrecover, keccak256, rlp, toBuffer } from 'ethereumjs-util'
+import { BaseTransaction } from './baseTransaction'
+import {
+  AccessList,
+  AccessListBuffer,
+  FeeMarketEIP1559TxData,
+  FeeMarketEIP1559ValuesArray,
+  JsonTx,
+  N_DIV_2,
+  TxOptions,
+} from './types'
+import { AccessLists } from './util'
+
+const TRANSACTION_TYPE = 2
+const TRANSACTION_TYPE_BUFFER = Buffer.from(TRANSACTION_TYPE.toString(16).padStart(2, '0'), 'hex')
+
+export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMarketEIP1559Transaction> {
+  public readonly chainId: BN
+  public readonly accessList: AccessListBuffer
+  public readonly AccessListJSON: AccessList
+  public readonly maxInclusionFeePerGas: BN
+  public readonly maxFeePerGas: BN
+
+  /**
+   * EIP-2930 alias for `r`
+   */
+  get senderR() {
+    return this.r
+  }
+
+  /**
+   * EIP-2930 alias for `s`
+   */
+  get senderS() {
+    return this.s
+  }
+
+  /**
+   * EIP-2930 alias for `v`
+   */
+  get yParity() {
+    return this.v
+  }
+
+  public static fromTxData(txData: FeeMarketEIP1559TxData, opts: TxOptions = {}) {
+    return new FeeMarketEIP1559Transaction(txData, opts)
+  }
+
+  /**
+   * Instantiate a transaction from the serialized tx.
+   *
+   * Note: this means that the Buffer should start with 0x01.
+   */
+  public static fromSerializedTx(serialized: Buffer, opts: TxOptions = {}) {
+    if (!serialized.slice(0, 1).equals(TRANSACTION_TYPE_BUFFER)) {
```

Comment:
nice check, i think we can add some coverage for this along with the other coverage check warnings below